src/v/wasm/probe.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "ssx/metrics.h"
+#include "utils/log_hist.h"
+
+#include <seastar/core/metrics.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/thread_cputime_clock.hh>
+#include <seastar/util/noncopyable_function.hh>
+
+#include <chrono>
+#include <memory>
+
+namespace wasm {
+
+// Per transform probe
+class transform_probe {
+public:
+    using hist_t = log_hist_public;
```

Comment:
is this histogram light enough to be used per transform ?

Reply:
`sizeof(log_hist_public) == 160` so I think so? If we have 1000 different transforms on a shard for 1000 partitions then we get 160kB which feels small enough. Happy to just switch this to a counter and have a stat for CPU utilization instead?